acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -5,24 +5,36 @@
 require 'openssl'
 require 'socket'
 require 'json'
+require 'securerandom'
 
 # This file contains general test helper methods for pxp-agent acceptance tests
 
 # The standard path for git checkouts is where pcp-broker will be
 GIT_CLONE_FOLDER = '/opt/puppet-git-repos'
 
 # Some helpers for working with the log file
-PXP_LOG_FILE_CYGPATH = '/cygdrive/c/ProgramData/PuppetLabs/pxp-agent/var/log/pxp-agent.log'
-PXP_LOG_FILE_POSIX = '/var/log/puppetlabs/pxp-agent/pxp-agent.log'
+PXP_LOG_FILE_CYGPATH = '/cygdrive/c/ProgramData/PuppetLabs/pxp-agent/var/log'
+PXP_LOG_FILE_POSIX = '/var/log/puppetlabs/pxp-agent'
 
-# @param host the beaker host you want the path to the log file on
-# @return The path to the log file on the host. For Windows, a cygpath is returned
-def logfile(host)
+def logfile_path(host)
   windows?(host)?
     PXP_LOG_FILE_CYGPATH :
     PXP_LOG_FILE_POSIX
 end
 
+# @param host the beaker host you want the path to the log file on
+# @return The path to the log file on the host. For Windows, a cygpath is returned
+def logfile(host)
+  logfile_path(host)+'/pxp-agent.log'
+end
+
+def reset_logfile(host)
+  old_logfile = logfile(host)
+  new_logfile = old_logfile+'.'+SecureRandom.uuid
+  logger.notify "---- Saving pxp-agent logfile to #{new_logfile}"
+  retry_on(host, "mv #{old_logfile} #{new_logfile}")
```

Comment:
I wonder if there's any potential test failure cases where old_logfile does not exist - that would cause a failure here.
I think the logfile can be missing if pxp-agent (the executable) exits very early.